### DISCUSSION TITLE
How to move an item

### DISCUSSION THREAD

**Discussion by matz33** [olsh/todoist-net#28] [Q&A] [answered]
Hi,

I would like to Move an existing item to another project. I assume I should use the method "MoveAsync". Its argument is an ItemMoveArgument, but the ProjectId of this argument is ReadOnly.

There is the class SectionMoveArgument also, where the New method allow to create it with Ids, but it can not be the argument of MoveAsync.

How can I move an item from one Project to another one ?

Thanks
Matz

**Comment by olsh** [comment, marked as answer]
Hi @matz33 

You should use `ItemMoveArgument.CreateMoveToProject` to create the `ItemMoveArgument` instance.
```csharp
  // Create two projects
  var firstProjectId = await client.Projects.AddAsync(new Project("First"));
  var secondProjectId = await client.Projects.AddAsync(new Project("Second"));

  // Create an item in the first project
  var itemId = await client.Items.AddAsync(new Item("Test task", firstProjectId));

  // Move the item to the second project
  await client.Items.MoveAsync(ItemMoveArgument.CreateMoveToProject(itemId, secondProjectId));
```

Also, the `ItemMoveArgument` has other static methods for other cases.

**Reply by matz33** [reply]
It works ! Thanks a lot, and thanks for this project that is very useful to me
